google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsTracerFactory.java

```diff
@@ -30,21 +46,75 @@
+              0.0, 0.01, 0.05, 0.1, 0.3, 0.6, 0.8, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 8.0, 10.0, 13.0,
+              16.0, 20.0, 25.0, 30.0, 40.0, 50.0, 65.0, 80.0, 100.0, 130.0, 160.0, 200.0, 250.0,
+              300.0, 400.0, 500.0, 650.0, 800.0, 1000.0, 2000.0, 5000.0, 10000.0, 20000.0, 50000.0,
+              100000.0));
+
+  public static final Aggregation AGGREGATION_COUNT =
+      Aggregation.explicitBucketHistogram(
+          ImmutableList.of(
+              1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 15.0, 20.0, 30.0, 40.0, 50.0,
+              100.0));
+
+  private static final String SCOPE = "bigtable.googleapis.com";
+  private final Attributes attributes;
+
+  private final BigtableInstruments bigtableInstruments;
 
-  public static BuiltinMetricsTracerFactory create(ImmutableMap<String, String> statsAttributes) {
-    return new BuiltinMetricsTracerFactory(statsAttributes);
+  public static BuiltinMetricsTracerFactory create(EnhancedBigtableStubSettings settings)
+      throws IOException {
+    return new BuiltinMetricsTracerFactory(settings);
   }
 
-  private BuiltinMetricsTracerFactory(ImmutableMap<String, String> statsAttributes) {
-    this.statsAttributes = statsAttributes;
+  BuiltinMetricsTracerFactory(EnhancedBigtableStubSettings settings) throws IOException {
+    this.attributes =
+        Attributes.builder()
+            .put(PROJECT_ID, settings.getProjectId())
+            .put(INSTANCE_ID, settings.getInstanceId())
+            .put(APP_PROFILE, settings.getAppProfileId())
+            .build();
+
+    if (settings.isBuiltinMetricsEnabled()) {
+      Resource resource = Resource.create(attributes);
+      MetricExporter metricExporter =
+          new BigtableCloudMonitoringExporter(settings.getCredentialsProvider().getCredentials());
+      InstrumentSelector histogramSelector =
+          InstrumentSelector.builder()
+              .setMeterName(SCOPE)
+              .setType(InstrumentType.HISTOGRAM)
+              .setUnit("ms")
+              .build();
+      InstrumentSelector counterSelector =
+          InstrumentSelector.builder()
+              .setMeterName(SCOPE)
+              .setType(InstrumentType.COUNTER)
+              .setUnit("1")
+              .build();
+
+      View histogramView = View.builder().setAggregation(AGGREGATION_WITH_MILLIS_HISTOGRAM).build();
+      View counterView = View.builder().setAggregation(Aggregation.sum()).build();
+
+      SdkMeterProvider meterProvider =
+          SdkMeterProvider.builder()
+              .setResource(resource)
+              .registerMetricReader(PeriodicMetricReader.create(metricExporter))
+              .registerView(histogramSelector, histogramView)
+              .registerView(counterSelector, counterView)
+              .build();
+      OpenTelemetry openTelemetry =
+          OpenTelemetrySdk.builder().setMeterProvider(meterProvider).build();
+      bigtableInstruments = new BuiltinInstruments(openTelemetry.getMeter(SCOPE));
```

Comment:
FYI, the meter name will be used for label `instrumentation_source`, which I think is a good fit for the CLIENT_NAME `java-bigtable`. We can also add the library version as `instrumentation_version` like [here](https://github.com/googleapis/sdk-platform-java/blob/6671379ed084889755401d5ab1ef7daa9a428728/gax-java/gax/src/main/java/com/google/api/gax/tracing/OpenTelemetryTracerFactory.java#L50), see the [example](https://opentelemetry.io/docs/instrumentation/java/manual/#metrics) from Otel for details.